🐛[BUG]: Cannot find module 'history' or its corresponding type declarations.

<!--
提问前先看看：

https://github.com/ryanhanwu/How-To-Ask-Questions-The-Smart-Way/blob/main/README-zh_CN.md
-->

### 🐛 bug 描述

When building a project with `@ant-design/pro-components`, we are getting `Cannot find module 'history' or its corresponding type declarations.`

<!--
详细地描述 bug，让大家都能理解
-->

### 📷 复现步骤
```bash
git clone https://github.com/maifeeulasad/react-pipeline.git
git checkout 3d5c8aed68657d576ee14ce65315a37284eec3e8
yarn
yarn build // <-- this will cause you error
```

Running this application in dev mode doesn't cause any trouble (`yarn start`).

Full error message:
```
node_modules/@ant-design/pro-layout/es/utils/getBreadcrumbProps.d.ts:2:20 - error TS2307: Cannot find module 'history' or its corresponding type declarations.

2 import type H from 'history';
                     ~~~~~~~~~
```

<!--
清晰描述复现步骤，让别人也能看到问题，如果可能，尽量提供可执行代码，
如：https://codesandbox.io/ 在此处创建一个 codesandbox，方便我们更快的排查和复现问题
-->

### 🏞 期望结果

It should have built without issues

<!--
描述你原本期望看到的结果
-->

### 💻 复现代码

<!--
提供可复现的代码，仓库，或线上示例
-->

### © 版本信息

- ProComponents 版本: `^2.6.34`
- umi 版本 `n/a`
- 浏览器环境 `Google Chrome:  118.0.5993.118 (Official Build) (64-bit)`
- 开发环境 `Windows / WSL2 / Ubuntu 22.04`

### 🚑 其他信息

 - Related issue: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/50526#issue-783271331
 - This can also be resolved by setting `"skipLibCheck": true,` for typescript compiler

<!--
如截图等其他信息可以贴在这里
-->

--- a/package.json
+++ b/package.json
@@ -59,11 +59,6 @@
   "browserslist": [
     "defaults"
   ],
-  "resolutions": {
-    "@types/react": "^18.0.38",
-    "@types/react-dom": "^18.0.11",
-    "@types/responselike": "^1.0.0"
-  },
   "dependencies": {
     "@ant-design/cssinjs": "^2.1.0",
     "@ant-design/icons": "^6.1.0",
@@ -107,7 +102,6 @@
     "@testing-library/react-hooks": "^8.0.1",
     "@testing-library/user-event": "^14.6.1",
     "@types/glob": "^8.1.0",
-    "@types/history": "^4.7.11",
     "@types/lodash-es": "^4.17.12",
     "@types/mockjs": "^1.0.10",
     "@types/node": "^14.18.63",

--- a/src/layout/utils/getBreadcrumbProps.tsx
+++ b/src/layout/utils/getBreadcrumbProps.tsx
@@ -3,7 +3,6 @@ import type {
   BreadcrumbItemType,
   ItemType,
 } from 'antd/lib/breadcrumb/Breadcrumb';
-import type H from 'history';
 import { match } from 'path-to-regexp';
 import type { ProSettings } from '../defaultSettings';
 import type { ProLayoutProps } from '../ProLayout';
@@ -14,7 +13,7 @@ export type BreadcrumbProLayoutProps = {
   breadcrumbList?: { title: string; href: string }[];
   home?: string;
   location?:
-    | H.Location
+    | Location
     | {
         pathname?: string;
       };

